src/batchnorm.jl

```diff
@@ -0,0 +1,485 @@
+"""
+
+`BNMoments` is a high-level data structure used to store running mean and running variance
+of batch normalization.
+
+# Fields
+ `momentum::AbstractFloat`: A real number between 0 and 1 to be used as the scale of
+  last mean and variance. The existing running mean or variance is multiplied with 
+  (1-momentum).
+ 
+ `mean`: The running mean.
+
+ `var`: The running variance.
+ 
+ `meaninit`: The function used for initialize the running mean. Should either be `nothing` or
+of the form `(eltype, dims...) -> data`. `zeros` is a good option.
+
+ `varinit`: The function used for initialize the running variance. Should either be `nothing` or
+`(eltype, dims...) -> data`. `ones` is a good option.
```

Comment:
Doesn't it take a long time to converge to actual values if we use ones and zeros?  I don't know if this is a problem in practice, but if it is we could try to set the mean and variance from the first few actual examples.

Reply:
In batchnorm, people often assume everything is a unit gaussian initially. I saw `ones` is the default initialization in other frameworks like [TF](https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization) and [PyTorch](http://pytorch.org/docs/master/_modules/torch/nn/modules/batchnorm.html), so I followed this convention. I'm not sure making it `zeros` would change anything.